Sources/AsyncQueue/FIFOQueue.swift

```diff
@@ -85,7 +85,7 @@ public final class FIFOQueue: Sendable {
     ///   - isolatedActor: The actor within which the task is isolated.
     ///   - task: The task to enqueue.
     /// - Returns: The value returned from the enqueued task.
-    public func enqueueAndWait<ActorType: Actor, T>(on isolatedActor: isolated ActorType, _ task: @escaping @Sendable (isolated ActorType) async -> T) async -> T {
+    public func enqueueAndWait<ActorType: Actor, T: Sendable>(on isolatedActor: isolated ActorType, _ task: @escaping @Sendable (isolated ActorType) async -> T) async -> T {
```

Comment:
This `T: Sendable` conformance was only required on methods that execute on an isolated actor. The warning we saw by not having this conformance was:
```
Non-sendable type 'T' returned by implicitly asynchronous call to actor-isolated function cannot cross actor boundary
```

This warning makes sense: `T` is not `Sendable`, and therefore how could we be passing it across contexts. However, the fact that this warning only surfaced when we explicitly utilizing an isolated parameter seems like a bug – in all cases we are sending `T` from an actor's isolated context to the current context. So I made all return types `T` conform to `Sendable`.

Reply:
To make sure I follow, we got a warning in all cases except the ones where the `task` closure doesn't have an `isolated` parameter? Making the return type `Sendable` seems reasonable but I want to try to understand better which methods didn't have a warning so I can try to understand why some methods are warning differently than others.

Reply:
If I'm following correctly I think that these two methods on `FIFOQueue` didn't have warnings.

```swift
public func enqueueAndWait<T: Sendable>(_ task: @escaping @Sendable () async -> T) async -> T
public func enqueueAndWait<T: Sendable>(_ task: @escaping @Sendable () async throws -> T) async throws -> T
```

Is that right?

Reply:
> To make sure I follow, we got a warning in all cases except the ones where the task closure doesn't have an `isolated` parameter?

Correct!

> Making the return type `Sendable` seems reasonable but I want to try to understand better which methods didn't have a warning so I can try to understand why some methods are warning differently than others.

My best guess is that this is a compiler bug. In all the methods I marked we're crossing a task boundary: from the task that created the async for loop to the call-site. But only in the methods where we have an `isolated` parameter does the compiler notice that we're crossing a task boundary and therefore only then does it require `Sendable` conformance.

> Is that right?

That is absolutely right.